app/gather/assets/apps/survey/mask/SurveyConfig.jsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2019 by eHealth Africa : http://www.eHealthAfrica.org
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react'
+import { defineMessages, injectIntl } from 'react-intl'
+import { getLabelTree } from '../../utils/types'
+
+const createMessages = visualizations => {
+  const prefix = 'survey.dashboard.config'
+  const messages = visualizations.reduce(
+    (acc, curr) => ({
+      ...acc,
+      [curr]: { id: `${prefix}.${curr.replace(' ', '_')}`, defaultMessage: curr }
+    }),
+    {
+      button: { id: `${prefix}.button`, defaultMessage: 'Activate dashboard' },
+      dashboard: { id: `${prefix}.dashboard_column`, defaultMessage: 'Dashboard' }
+    }
+  )
+  return defineMessages(messages)
+}
+
+const SurveyConfig = ({
```

Comment:
One of these days you may explain to me why you don't like the `Component` class. :thinking:

Reply:
Functional Components are more easier to read, test and less code of course... And btw, i think i shouldn't inherit `React.Component` bunch of stuff I'll end up not using. I only need one piece of state to maintain 🤔.

Reply:
:+1: